### PR TITLE
httpx: add page

### PR DESCRIPTION
**Pull request by fazlearefin**
- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message-and-pr-title).
- **Version of the command being documented (if known):**

### CODE REVIEW

File: pages/common/httpx.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+- Run a probe against a list of hosts [f]iltering out [c]odes of certain responses:
+
+`httpx -u {{host1}},{{host2}},{{hostN}} -fc {{400,401,404}}`
+
+- Run a probe against a list of hosts [m]atching [c]odes of certain responses:
+
+`httpx -u {{host1}},{{host2}},{{hostN}} -mc {{200,301,304}}`
+
+- Run a probe against a URL [s]aving [s]creenshots of certain paths, with [s]creenshot [t]imeouts (assets are saved in ./output):
```
Comment (vitorhcl) [inline pages/common/httpx.md:35]:
```suggestion
- Run a probe against a URL [s]aving [s]creenshots of certain paths, with [s]creenshot [t]imeouts (assets are saved in `./output`):
```
From the [style guide](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#special-cases):
> Technical wording on description lines should use the `backtick` syntax.
> Use backticks on the following:
> 
> - Paths, e.g. `package.json`, `/etc/package.json`.
> - Extensions, e.g. `.dll`.
> - Commands, e.g. `ls`.
> - Standard streams: `stdout`, `stdin`, `stderr`. **Do not** use the full names (e.g. standard output).

**Review by vitorhcl — CHANGES_REQUESTED** [review]
Thank you for this excellent PR! I have some suggestions

File: pages/common/httpx.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+`httpx -rl {{150}} -l {{file_containing_newline_separate_hosts}} -td -rt`
+
+- Run a probe against a URL showing webpage title, CDN/WAF in use, and page content hash:
+
+`httpx -u {{url}} -title -cdn -hash {{sha256}}`
+
+- Run a probe against a list of hosts with custom defined [p]orts and timeout after certain seconds:
+
+`httpx -probe -u {{host1}},{{host2}},{{hostN}} -p http:{{80,8000-8080}},https:{{443,8443}} -timeout {{10}}`
```
[the remark this answers is not in the document]
Comment (fazlearefin) [reply pages/common/httpx.md:25]: Wrong syntax if used in `httpx` as hosts must be separated by commas. Here is the output from `httpx`

```console
# WITH SPACE SEPARATED, ONLY THE FIRST HOST IS SCANNED

$ httpx -u www.google.com www.yahoo.com www.github.com -p http:80,81,https:443,8443 -timeout 5

    __    __  __       _  __
   / /_  / /_/ /_____ | |/ /
  / __ \/ __/ __/ __ \|   /
 / / / / /_/ /_/ /_/ /   |
/_/ /_/\__/\__/ .___/_/|_|
             /_/

                projectdiscovery.io

[INF] Current httpx version v1.4.0 (latest)
https://www.google.com
$

###############################################################

# USING THE CORRECT SYNTAX ALL HOSTS ARE SCANNED

$ httpx -u www.google.com,www.yahoo.com,www.github.com -p http:80,81,https:443,8443 -timeout 5 

    __    __  __       _  __
   / /_  / /_/ /_____ | |/ /
  / __ \/ __/ __/ __ \|   /
 / / / / /_/ /_/ /_/ /   |
/_/ /_/\__/\__/ .___/_/|_|
             /_/

                projectdiscovery.io

[INF] Current httpx version v1.4.0 (latest)
http://www.github.com
https://www.github.com
https://www.yahoo.com
http://www.google.com
https://www.google.com
http://www.yahoo.com
$ 
```

If we must follow the style guide here, maybe this is acceptable?

```console
`httpx -probe -u {{host1,host2,...}} -p http:{{80,8000-8080}},https:{{443,8443}} -timeout {{10}}`
```

File: pages/common/httpx.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+- Run a probe against a list of hosts [f]iltering out [c]odes of certain responses:
+
+`httpx -u {{host1}},{{host2}},{{hostN}} -fc {{400,401,404}}`
```
[the remark this answers is not in the document]
Comment (fazlearefin) [reply pages/common/httpx.md:29]: See above comment.

File: pages/common/httpx.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+- Run a probe against a list of hosts [m]atching [c]odes of certain responses:
+
+`httpx -u {{host1}},{{host2}},{{hostN}} -mc {{200,301,304}}`
```
[the remark this answers is not in the document]
Comment (fazlearefin) [reply pages/common/httpx.md:33]: See above comment.

File: pages/common/httpx.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (vitorhcl) [reply pages/common/httpx.md:25]: Oh, sorry, overlooked it, but yeah, this suggestion fits the style guide just fine.

[the remark this answers is not in the document]
Comment (fazlearefin) [reply pages/common/httpx.md:25]: I will push the changes following the style guide.

[the remark this answers is not in the document]
Comment (fazlearefin) [reply pages/common/httpx.md]: Fixed

File: pages/common/httpx.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (fazlearefin) [reply pages/common/httpx.md]: Fixed

File: pages/common/httpx.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (fazlearefin) [reply pages/common/httpx.md]: Fixed

**Review by kbdharun — APPROVED** [review]

File: pages/common/httpx.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# httpx
+
+> A fast and multi-purpose HTTP toolkit written in Go to run multiple probes at once.
+> (Note that this is not the Python HTTPX which has the exact same command name).
```
Comment (fazlearefin) [inline pages/common/httpx.md:4]:
```suggestion
> Note: This page uses the same command name as the Python HTTPX library, but it's not related.
```

[the remark this answers is not in the document]
Comment (fazlearefin) [reply pages/common/httpx.md:4]: I suggested an alternate suggestion below which I think is provides more clarity.

Comment (vitorhcl) [reply pages/common/httpx.md:4]:
```suggestion
> Note: not to be confused with Python HTTPX which has the exact same command name.
```
How about this?

**Review by gutjuri — APPROVED** [review]

**Review by kbdharun — APPROVED** [review]

**Review by vitorhcl — APPROVED** [review]

**Pull request merged**